scripts/allocate_ip.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+#
+# This script is expected to run on the admin (crowbar) server and will add connect a given
+# host to a given network on a particular range. The script is called with:
+#
+#    allocate_ip.sh <node> <network> <range>
+#
+# As an example:
+#
+#    allocate_ip.sh d52-54-77-77-01-03.virtual.cloud.suse.de storage host
+#
+# Would add node d52-54-77-77-01-03.virtual.cloud.suse.de as a host to the storage network.
+#
+# This script is used in the function onadmin_external_ceph by qa_crowbarsetup.sh as apart of
+# the deployexternalceph step to make sure all ceph cluster nodes are added to the correct ceph
+# public and cluster networks.
+set -eu
+
+#Specify full node name as first parameter ~ d52-54-77-77-01-01.virtual.cloud.suse.de
+NODE_NAME="$1"
+
+#Specify network as second parameter
+NETWORK_TO_ALLOCATE="$2"
+
+#Specify range as third parameter
+RANGE_TO_ALLOCATE="$3"
+
+TEMPSCRIPT=$(mktemp)
+
+function remove_tempfile() {
+    rm -f $TEMPSCRIPT
+}
+trap remove_tempfile EXIT
+chown crowbar $TEMPSCRIPT
+
+cat >$TEMPSCRIPT << EOF
+net_svc = NetworkService.new Chef::Log
+ip_addr = net_svc.allocate_ip "default", "$NETWORK_TO_ALLOCATE", "$RANGE_TO_ALLOCATE", "$NODE_NAME"
```

Comment:
there is probably something obvious I'm missing here, but why are you not simply using crowbar(ctl) for allocating an ip address to a host?

Reply:
That is a great question. The script was written by Mate. So I've just had a play and looks like I could:
  crowbarctl network hostip allocate default <node> <network> host

I'd still need to ssh into the node to run chef-client so the network updates. Unless there is a better way. I'll change the qa_crowbar step and see how it goes